### ISSUE TITLE
sf project deploy start --verbose doesn't display errors like sfdx force:source:deploy --verbose does

### ISSUE THREAD

**Issue by achereshnevdg** [forcedotcom/cli#2166]
### Summary

sf project deploy start just silently fails while sf force:source:deploy provides insight into the error while being passed a --verbose flag

### Steps To Reproduce:

**Repository to reproduce:** [dreamhouse-lwc](https://github.com/dreamhouseapp/dreamhouse-lwc)

***NOTE:** If your issue is not reproducable by dreamhouse-lwc, i.e. requires specific metadata or files, we **require** a link to a simple Salesforce project repository with a script to setup a scratch org that reproduces your problem.*

1. Add a package.xml file
2. Specify a non-existent metadata:
```
<?xml version="1.0" encoding="UTF-8"?>
<Package xmlns="http://soap.sforce.com/2006/04/metadata">
    <types>
        <members>NonExistentObject__c</members>
        <name>CustomObject</name>
    </types>
    <version>56.0</version>
</Package>
```
3. Run `sf project deploy start -x package/package.xml --verbose -o <orgname> -c`

### Expected result

Component failures should be displayed like when you run `sfdx force:source:deploy -x package/package.xml --verbose -u <orgname> -c`

### Actual result

Component failures aren't displayed

### System Information

Shell doesn't matter

sf version
{
  "cliVersion": "@salesforce/cli/1.79.0",
  "architecture": "win32-x64",
  "nodeVersion": "node-v18.15.0",
  "osVersion": "Windows_NT 10.0.22621",
  "shell": "C:\\Program Files\\Git\\usr\\bin\\bash.exe",
  "rootPath": "C:\\Users\\alexander.chereshnev\\AppData\\Local\\sfdx\\client\\7.201.6-4edd82d\\sf",
  "pluginVersions": [
    "@oclif/plugin-autocomplete 2.2.0 (core)",
    "@oclif/plugin-commands 2.2.15 (core)",
    "@oclif/plugin-help 5.2.9 (core)",
    "@oclif/plugin-not-found 2.3.23 (core)",
    "@oclif/plugin-plugins 3.0.1 (core)",
    "@oclif/plugin-search 0.0.17 (core)",
    "@oclif/plugin-update 3.1.14 (core)",
    "@oclif/plugin-version 1.3.4 (core)",
    "@oclif/plugin-warn-if-update-available 2.0.36 (core)",
    "@oclif/plugin-which 2.2.20 (core)",
    "@salesforce/cli 1.79.0 (core)",
    "apex 2.2.17 (core)",
    "auth 2.7.15 (core)",
    "data 2.3.16 (core)",
    "deploy-retrieve 1.8.20 (core)",
    "functions 1.21.7 (user)",
    "info 2.6.10 (core)",
    "limits 2.3.14 (core)",
    "login 1.2.8 (core)",
    "org 2.8.2 (core)",
    "packaging 1.16.10 (user)",
    "schema 2.3.10 (core)",
    "settings 1.4.6 (core)",
    "sobject 0.1.18 (core)",
    "source 2.10.7 (core)",
    "telemetry 2.2.0 (core)",
    "templates 55.4.13 (core)",
    "trust 2.4.16 (core)",
    "user 2.3.10 (core)"
  ]
}

**Comment by iowillhoit** [comment]
Hello @achereshnevdg, I believe this was recently fixed in [this PR](https://github.com/salesforcecli/plugin-deploy-retrieve/pull/623). It should be included in `plugin-deploy-retrieve` version `1.10.3`.  You can give it a try by installing the nightly of `sf`. `npm install @salesforce/cli@nightly --global`

**Comment by cristiand391** [comment]
This was fixed in sf v1.81.0, release notes:
https://github.com/forcedotcom/cli/tree/main/releasenotes/sf#1810-june-7-2023-stable
